pkg/provision/storage/shared.go

```diff
@@ -277,3 +277,32 @@ func checkPVCTerminating(name, namespace string, api sync.ClusterAPI) (bool, err
 	}
 	return pvc.DeletionTimestamp != nil, nil
 }
+
+func shouldPersistHomeDirectory(workspace *common.DevWorkspaceWithConfig) bool {
+	return workspace.Config.Workspace.PersistUserHome != nil && workspace.Config.Workspace.PersistUserHome.Enabled != nil && *workspace.Config.Workspace.PersistUserHome.Enabled
+}
+
+// Adds a home volume mount to the given set of Pod Additions.
+// The home volume mount is mounted to the given PVC at the path `/home/user/`.
+// If a Devfile volume already mounts to `/home/user/`, an error is returned.
+func addHomeVolumeMount(workspace *common.DevWorkspaceWithConfig, pvcName string, podAdditions *v1alpha1.PodAdditions) error {
+	homeUserDirPath := "/home/user/"
+	homeVolumeMount := corev1.VolumeMount{
+		Name:      pvcName,
+		MountPath: homeUserDirPath,
+		SubPath:   fmt.Sprintf("%s/home/user/", workspace.Status.DevWorkspaceId),
+	}
+
+	// Check if we are already trying to mount a volume to /home/user/ in one of the containers
+	for cIdx := range podAdditions.Containers {
+		for vmIdx := range podAdditions.Containers[cIdx].VolumeMounts {
+			if podAdditions.Containers[cIdx].VolumeMounts[vmIdx].MountPath == homeUserDirPath {
+				containerName := podAdditions.Containers[cIdx].Name
+				volumeMountName := podAdditions.Containers[cIdx].VolumeMounts[vmIdx].Name
+				return fmt.Errorf("volumeMount '%s' for container '%s' already mounts to %s", volumeMountName, containerName, homeUserDirPath)
```

Comment:
> I'm kind of partial to no. 2 above -- if someone wants to configure it themself we shouldn't necessarily mess with that.

Agreed, I ended up changing things so that if a user's devfile already has a volume mounting to `/home/user`, the persistentHome volume is not added and the feature is essentially disabled for that workspace.